shutdown.go

```diff
@@ -0,0 +1,65 @@
+package gokit
```

Comment:
i think the name of this file is a bit strange. Why not call it server or gracefulserver or httpserver?

Reply:
https://golang.org/doc/effective_go.html#package-names
https://talks.golang.org/2014/names.slide#17

To name the package to what you want, I need to de-link it from the repo, creating another folder. But I don't think it should be called the names above as the package will become `server.Server`, `gracefulserver.Server` and `httpserver.Server`.

Instead, I propose calling it `web` like what ArdanLabs did for its `kit`, then the struct will be called `web.Server`

Reply:
I think the package name `web` is fine but the filename would make more sense to be called `server.go` instead of `shutdown.go`.

Reply:
Sure, why not.